src/internal/operations/backup.go

```diff
@@ -312,6 +318,10 @@ func (op *BackupOperation) do(
 		lastBackupVersion = mans.MinBackupVersion()
 	}
 
+	// TODO(ashmrtn): This should probably just return a collection that deletes
+	// the entire subtree instead of returning an additional bool. That way base
+	// selection is controlled completely by flags and merging is controlled
+	// completely by collections.
```

Comment:
no, it should be safe due to how we select bases and persist data. The "deletion" happens during hierarchy merging so it's really just excluding directories from the base when building the in-memory state

If corso crashes partway through the item data backup with the "deleted" state in-memory then we won't select that as a base with the new corso extension base selection logic (it won't have a details snapshot)

If corso hits some error persisting item data while making the snapshot but does still persist the details then the base will be eligible as an assist base on the next backup. That's safe though because the assist base contains all the details we'll need and the original merge base still has all the undeleted items